src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -549,90 +627,133 @@ def _get_enable_managed_identity(
+                enable_pod_identity = self.mc.pod_identity_profile.enabled
 
         # this parameter does not need dynamic completion
         # validation
         if enable_validation:
-            if enable_pod_identity and not self._get_enable_managed_identity(enable_validation=False):
-                raise RequiredArgumentMissingError(
-                    "--enable-pod-identity can only be specified when --enable-managed-identity is specified"
-                )
-            if self.mc and self.mc.network_profile and safe_lower(self.mc.network_profile.network_plugin) == "kubenet":
-                if enable_pod_identity and not self._get_enable_pod_identity_with_kubenet(enable_validation=False):
+            if self.decorator_mode == DecoratorMode.CREATE:
+                if enable_pod_identity and not self._get_enable_managed_identity(enable_validation=False):
                     raise RequiredArgumentMissingError(
-                        "--enable-pod-identity-with-kubenet is required for enabling pod identity addon "
-                        "when using Kubenet network plugin"
+                        "--enable-pod-identity can only be specified when --enable-managed-identity is specified"
+                    )
+                self.validate_pod_identity_with_kubenet(
+                    self.mc,
+                    enable_pod_identity,
+                    self._get_enable_pod_identity_with_kubenet(
+                        enable_validation=False
+                    ),
+                )
+            elif self.decorator_mode == DecoratorMode.UPDATE:
+                if enable_pod_identity and self._get_disable_pod_identity(enable_validation=False):
+                    raise MutuallyExclusiveArgumentError(
+                        "Cannot specify --enable-pod-identity and "
+                        "--disable-pod-identity at the same time."
                     )
         return enable_pod_identity
 
     def get_enable_pod_identity(self) -> bool:
         """Obtain the value of enable_pod_identity.
 
         This function will verify the parameter by default. If enable_managed_identity is not specified but
-        enable_pod_identity is, raise a RequiredArgumentMissingError. If network_profile has been set up in `mc`,
-        network_plugin equals to "kubenet" and enable_pod_identity is specified but enable_pod_identity_with_kubenet
-        is not, raise a RequiredArgumentMissingError.
+        enable_pod_identity is, raise a RequiredArgumentMissingError. Will also call function
+        "validate_pod_identity_with_kubenet" for verification. In update mode, if both enable_pod_identity and
+        disable_pod_identity are specified, raise a MutuallyExclusiveArgumentError.
 
         :return: bool
         """
 
         return self._get_enable_pod_identity(enable_validation=True)
 
+    # pylint: disable=unused-argument
+    def _get_disable_pod_identity(self, enable_validation: bool = False, **kwargs) -> bool:
```

Comment:
question: why do we want to use `**kwargs` for this helper method? Is this a convention or just a catch all usage? I think it's better to explicitly specify the parameter we want to use here.

Reply:
I've added this additional parameter `**kwargs` for all the internal version of the getter functions. The original intention is to pass additional parameters in some special cases, but I have not used it so far. For the getter functions with complex logic, I have prepared both "public" function and "internal" function (function name starts with `_`) for them.
- The "public" function basically has no parameter and should be used in the decorator class (e.g. `AKSCreateDecorator`/`AKSPreviewCreateDecorator`), where users do not need to consider the details of getter implementation.
- The "internal" function has many parameters, which are used to control whether to perform validation or dynamic completion, etc. It should be used in the context class (e.g. `AKSContext`/`AKSPreviewContext`). With proper parameter settings, infinite loop calls during cross-checking would be avoided.

Reply:
> The original intention is to pass additional parameters in some special cases, but I have not used it so far

I don't this is the pythonic way. By adding `**kwargs` here it's easy to invite people to put many things inside it, which eventually become a large bag without typing information. This would be hard to refactor and test in the future. I think this is also the case before your refactor - we don't know what are we referencing in each sub command.

> With proper parameter settings, infinite loop calls during cross-checking would be avoided.

Could you provide examples? I think if calling without setting special parameter settings could cause inf. loop, that means something wrong in the code path because it's easy to write it wrong. We should properly rethink the design.

Reply:
For the first part, I agree with you and I would remove the `**kwargs`.

For the second question, take the `enable_validation` parameter as an example, the default value is `False`, which means if you call the internal function without any parameter, it will skip the validation part and this would avoid loop calls.

So don’t worry, for almost all situations, we could avoid loop calls by the default values. The situations that require additional parameter settings are usually related to dynamic completion. At present, all these cases are in azure-cli. Some of the parameters involved are sp and managed identity, fqdn subdomain and dns prefix.

> For a getter function, the internal is mainly composed of three parts, obtaining the value, dynamic modification, and verification. In the verification part, we conduct cross-validation between multiple parameters (the verification of a single parameter is handled by the validator (defined in `_validators.py`) specified in the parameter declaration (defined in `_params.py`)). Take the getters for `enable_pod_identity` and `disable_pod_identity` as the example, When verifying whether `disable_pod_identity` is set at the same time in the getter of `enable_pod_identity`, we will call the getter of `disable_pod_identity` to obtain its value, and the verification part of `disable_pod_identity` will also obtain the value of `enable_pod_identity` by calling its getter to verify whether the two are assigned at the same time. If we do not use a "flag" to escape this verification, then we will fall into a loop call.